### PR TITLE
deprecate 0.11 syntax in docs (resources)

### PR DESCRIPTION
**Pull request by megan07**
Changes of upgrading to 0.12 syntax for resource hand-written docs.

I also updated the suffixes of a few `.md` files and `.markdown` files to be consistent with `.html.markdown`

### CODE REVIEW

File: third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 1]
```diff
@@ -100,17 +104,29 @@ resource "random_id" "db_name_suffix" {
...
+        }
+      }
+      dynamic "authorized_networks" {
+        for_each = [data.null_data_source.auth_netw_postgres_allowed_2.*.outputs]
+        content {
+          expiration_time = lookup(authorized_networks.value, "expiration_time", null)
+          name            = lookup(authorized_networks.value, "name", null)
+          value           = lookup(authorized_networks.value, "value", null)
+        }
+      }
```
Comment (megan07) [inline third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: `terraform 0.12upgrade` left this comment here:

>  TF-UPGRADE-TODO: The automatic upgrade tool can't predict
>  which keys might be set in maps assigned here, so it has
>  produced a comprehensive set here. Consider simplifying
>  this after confirming which keys can be set in practice.

After looking above do we not need `expiration_time` set in these?

File: third_party/terraform/website/docs/r/dataproc_cluster.html.markdown [hunk 2]
```diff
@@ -20,100 +20,94 @@ whole cluster!
...
+    # Override or set some custom properties
+    software_config {
+      image_version = "1.3.7-deb9"
+      override_properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+      }
+    }
+
+    gce_cluster_config {
+      #network = "${google_compute_network.dataproc_network.name}"
```
Comment (chrisst) [inline third_party/terraform/website/docs/r/dataproc_cluster.html.markdown]: remove?

File: third_party/terraform/website/docs/r/sql_database_instance.html.markdown (hunk 1 above)
Comment (chrisst) [reply third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: I'm not too familiar with this example, but from my reading the goal is to create a list of IP addresses including all of the compute_instance ip's and the [two manual ones](https://github.com/GoogleCloudPlatform/magic-modules/pull/2574/files#diff-18e82d50a24d79aee2e6bd023c4b3905R98). Then make sure that all of those ip's are listed as authorized networks and throws a `name` in there for good measure. So I feel pretty confident that `expiration_time` is not needed.
However I'm less confident about the the dynamic block syntax, but it seems like we could get away with removing the null resources here completely and accomplish the same goal. Which I think would be a great example of "here's something we had to work around in 0.11 that isn't such a hack in 0.12".

File: third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 3]
```diff
@@ -111,20 +94,24 @@ resource "google_sql_database_instance" "postgres" {
...
       dynamic "authorized_networks" {
-        for_each = [data.null_data_source.auth_netw_postgres_allowed_2.*.outputs]
+        for_each = local.onprem
+        iterator = onprem
+
         content {
-          expiration_time = lookup(authorized_networks.value, "expiration_time", null)
-          name            = lookup(authorized_networks.value, "name", null)
-          value           = lookup(authorized_networks.value, "value", null)
+          name  = "onprem-${onprem.key}"
```
Comment (chrisst) [inline third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: I'm assuming this is the index?

File: third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 4]
```diff
@@ -77,40 +78,42 @@ resource "google_compute_instance" "apps" {
...
   settings {
     tier = "db-f1-micro"
 
     ip_configuration {
-      authorized_networks = [
-        "${data.null_data_source.auth_netw_postgres_allowed_1.*.outputs}",
-        "${data.null_data_source.auth_netw_postgres_allowed_2.*.outputs}",
-      ]
+
+      dynamic "authorized_networks" {
```
Comment (chrisst) [inline third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: Looks great! Way nicer than the workaround with null resources.

File: third_party/terraform/website/docs/r/sql_database_instance.html.markdown (hunk 3 above)
Comment (megan07) [reply third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: yup!

**Pull request merged**